### ISSUE TITLE
Cannot use lodash/get in solid-start v0.1.2

### ISSUE THREAD

**Issue by patdx** [solidjs/solid-start#297]
Reproduction: https://github.com/patdx/solid-lodash-test/blob/55575cd0ce492a17fa062d8551d1659bbafb9e21/src/routes/index.jsx#L3

Open reproduction in Stackblitz: https://stackblitz.com/github/patdx/solid-lodash-test

When I run `pnpm dev` it seems to run on the server side but the browser side shows a console error:

![image](https://user-images.githubusercontent.com/353863/193746310-8680b44b-08fb-4453-893f-11deb09ca2f9.png)

And vite is serving a commonjs module (`module.exports = ...`) to the browser instead of transforming it into ESM:

![image](https://user-images.githubusercontent.com/353863/193746389-ae620df7-43b8-4dcf-90e9-23d8fbb71ce7.png)

It works in solid-start v0.1.1 and in a plain Vite Solid app with `vite-plugin-solid: 2.3.9`. Based on some quick experiments, maybe caused by the `optimizeDeps.disabled: true` that was recently added? https://github.com/solidjs/solid-start/blob/e94ea868216b5f475ca5f9e7034fbe1d9fbb6f64/packages/start/vite/plugin.js#L496

I understand that there is some kind of solid library optimization issue that is perhaps blocking this issue, but I am afraid I do not understand it completely.

**Comment by ryansolid** [comment]
Yeah.

At some point recently Vite started creating different hashes for node modules. This would break things like `server$` functions in dev. The only way to avoid it was to add extensions for `.jsx` to optimizeDeps. But we don't really want anything with `.jsx` to make it to ESBuild. We tried to be smart about exclusion but there were gaps so we thought we could just turn it off. But this issue suggests we can't.

Basically every variation of config produces a different issue for different packages at this point.

**Comment by patdx** [comment]
Thanks. I understand the issue a little bit better. Is the reason to tweak with the optimizeDep settings to try to force the solid components in dependencies to be transformed? Hypothetically speaking, could it help to have a proper esbuild plugin? Something like this?

```js
export const config = defineConfig({
  optimizeDeps: {
    extensions: ['jsx', 'tsx'],
    esbuildOptions: {
      plugins: [solidEsbuildPlugin()],
      conditions: ['solid', '...'],
    },
  },
  plugins: [solidVitePlugin()],
});
```

**Comment by nksaraf** [comment]
Yeah this is going to be the actual solution I think. You are right on. We were just hoping we didn't have to do this. But it might be time to just put our helps down and try to get this working.

**Comment by patdx** [comment]
Thanks, can confirm things are working nicely for me for my little reproduction https://github.com/patdx/solid-lodash-test once I upgraded to solid v1.5.7 and solid-start v0.1.3.

if this general strategy is working, to make the long term maintenance easy, it may be nice to have the esbuild and vite plugin deployed in the same npm package, to ensure that the `babel-plugin-jsx-dom-expressions` and Solid HMR versions are always in sync. I think [unplugin](https://github.com/unjs/unplugin) may be useful for this.

**Comment by ryansolid** [comment]
Yeah I imagine that we will just bring the esbuild plugin at least as a dep of vite-plugin-solid. We've just broken so many things messing with this wanted to make sure it was good.
